drawing_agent/app/nodes.py

```
@@ -1,124 +1,143 @@
+            content_text = m.content
+        elif isinstance(m.content, list):
+            # Извлекаем текст, игнорируя картинки для корректного подсчета токенов
+            parts = [str(item.get("text", "")) for item in m.content if item.get("type") == "text"]
+            content_text = " ".join(parts)
+
+        # Если контент пустой (например, только картинка без текста),
+        # подставляем заглушку, чтобы trim_messages не падал
+        if not content_text.strip():
+            content_text = "..."
+
+        # Создаем копию сообщения того же типа, но только с текстом
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `с` (CYRILLIC SMALL LETTER ES). Did you mean `c` (LATIN SMALL LETTER C)?

(RUF003)